app/backend/app.py

```diff
@@ -277,13 +277,122 @@ async def get_all_upload_status(request: Request):
     timeframe = json_body.get("timeframe")
     state = json_body.get("state")
+    folder = json_body.get("folder")
     try:
-        results = statusLog.read_files_status_by_timeframe(timeframe, State[state])
+        results = statusLog.read_files_status_by_timeframe(timeframe, State[state], 
+            folder, os.environ["AZURE_BLOB_STORAGE_UPLOAD_CONTAINER"])
+
+        # retrieve tags for each file
+         # Initialize an empty list to hold the tags
+        items = []              
+        cosmos_client = CosmosClient(url=tagsHelper._url, credential=tagsHelper._key)
+        database = cosmos_client.get_database_client(tagsHelper._database_name)
+        container = database.get_container_client(tagsHelper._container_name)
+        query_string = "SELECT DISTINCT VALUE t FROM c JOIN t IN c.tags"
+        items = list(container.query_items(
+            query=query_string,
+            enable_cross_partition_query=True
+        ))           
+
+        # Extract and split tags
+        unique_tags = set()
+        for item in items:
+            tags = item.split(',')
+            unique_tags.update(tags)        
+        
     except Exception as ex:
         log.exception("Exception in /getalluploadstatus")
         raise HTTPException(status_code=500, detail=str(ex)) from ex
     return results
 
+@app.post("/getfolders")
+async def get_folders(request: Request):
+    """
+    Get all folders.
+
+    Parameters:
+    - request: The HTTP request object.
+
+    Returns:
+    - results: list of unique folders.
+    """
+    try:
+        blob_container = blob_client.get_container_client(os.environ["AZURE_BLOB_STORAGE_UPLOAD_CONTAINER"])
+        # Initialize an empty list to hold the folder paths
+        folders = []
+        # List all blobs in the container
+        blob_list = blob_container.list_blobs()
+        # Iterate through the blobs and extract folder names and add unique values to the list
+        for blob in blob_list:
+            # Extract the folder path if exists
+            folder_path = os.path.dirname(blob.name)
+            if folder_path and folder_path not in folders:
+                folders.append(folder_path)
+    except Exception as ex:
+        log.exception("Exception in /getfolders")
+        raise HTTPException(status_code=500, detail=str(ex)) from ex
+    return folders
+
+
+@app.post("/deleteItems")
+async def delete_Items(request: Request):
+    """
+    Delete a blob.
+
+    Parameters:
+    - request: The HTTP request object.
+
+    Returns:
+    - results: list of unique folders.
+    """
+    json_body = await request.json()
+    path = json_body.get("path")
+    # remove the container prefix
+    path = path.split("/", 1)[1]
+    try:
+        blob_container = blob_client.get_container_client(os.environ["AZURE_BLOB_STORAGE_UPLOAD_CONTAINER"])
+        blob_container.delete_blob(path)
+
+    except Exception as ex:
+        log.exception("Exception in /getfolders")
```

Comment:
Potential copy-paste issue as this log.exception is in deleteItems. Please update the log message to read Exception in deleteItems